crates/simulate/src/stochastic/price_process.rs

```diff
@@ -0,0 +1,184 @@
+    /// Geometric Brownian Motion (GBM) process.
+    GBM,
+    /// Ornstein-Uhlenbeck (OU) process.
+    OU,
+}
+
+/// Struct for all price processes init parameters.
+/// A price process is a stochastic process that describes the evolution of a price.
+/// # Fields
+/// * `process_type` - Type of price process. (PriceProcessType)
+/// * `timestep` - Time step of the simulation. (f64)
+/// * `timescale` - Time in string interpretation. (String)
+/// * `num_steps` - Number of steps in the simulation. (usize)
+/// * `initial_price` - Initial price of the simulation. (f64)
+/// * `seed` - Seed for testing. (u64)
+pub struct Price {
+    pub process_type: PriceProcessType,
+    pub timestep: f64,
+    pub timescale: String,
+    pub num_steps: usize,
+    pub initial_price: f64,
+    pub seed: u64,
+}
+
+impl Price {
+    /// Public builder function that instantiates a [`Price`].
+    pub fn new(
+        process_type: PriceProcessType,
+        timestep: f64,
+        timescale: String,
+        num_steps: usize,
+        initial_price: f64,
+        seed: u64,
+    ) -> Self {
+        Price {
+            process_type,
+            timestep,
+            timescale,
+            num_steps,
+            initial_price,
+            seed,
+        }
+    }
+}
+
+impl Plot for Price {
+    /// Plots a price path vs time.
+    /// # Arguments
+    /// * `time` - Vector of time steps. (Vec<f64>)
+    /// * `price_path` - Vector of prices. (Vec<f64>)
+    /// # Returns
+    /// * `filename` - Name of the file. (String)
+    /// # Panics
+    /// * `PlottingPrice.html` - If the file cannot be created.
+    fn plot(&self, time: &[f64], price_path: &[f64]) {
+        let mut filename = String::from("PlottingPrice");
+        filename.push_str(".html");
+
+        let mut plot = Plot::new();
+        let trace = Scatter::new(time.to_owned(), price_path.to_owned());
+        plot.add_trace(trace);
+
+        plot.write_html(filename) // Produces .html using the identifier in arbiter root directory.
+    }
+}
+
+/// Geometric Brownian Motion process parameters struct.
+/// # Fields
+/// * `drift` - Price drift of the underlying asset. (f64)
+/// * `volatility` - Volatility of the underlying asset. (f64)
+struct GBM {
+    drift: f64,
+    volatility: f64,
+}
+
+impl GBM {
+    /// Public builder function that instantiates a [`GBM`].
+    pub fn new(drift: f64, volatility: f64) -> Self {
+        GBM { drift, volatility }
+    }
+    /// Generates a GBM price path.
+    /// # Arguments
+    /// * `timestep` - Time step of the simulation. (f64)
+    /// * `num_steps` - Number of steps in the simulation. (usize)
+    /// * `initial_price` - Initial price of the simulation. (f64)
+    /// * `seed` - Seed for testing. (u64)
+    /// # Returns
+    /// * `time` - Vector of time steps. (Vec<f64>)
+    /// * `prices` - Vector of prices. (Vec<f64>)
+    pub fn generateGBM(&self, timestep: f64, num_steps: usize, initial_price: f64, seed: u64) -> (Vec<f64>, Vec<f64>) {
```

Comment:
Are you planning to just pass a `Price` struct into this? That could be a good way to cut down on the number of inputs.

Reply:
yes I am! currently working on it along with integration, will push recent changes today